qt/python/mantidqt/project/encoderfactory.py

```diff
@@ -30,7 +32,12 @@ def find_encoder(cls, obj):
         elif len(obj_encoders) == 1:
             return obj_encoders[0]()
         else:
-            return None
+            # Search for a c++ interface if it's present
```

Comment:
As discussed offline - I think it should be possible to have the C++ return a pointer to the `BaseEncoder*` type but the solution of having the C++ side do the encode/decoding is fine and not worth changing at this point. The thing that I think we need to do here is separate the logic here out a bit as we have one case where you get the `Encoder` back and another case where you get the encoded object as a `dict` back. The asymmetry, entirely dependent on whether the interface is Python/C++ could get confusing to callers and will lead to checks everywhere on the return type.

I suggest we introduce a higher level `encode` function that basically encapsulates this bit and does the encoding so that you always get either a `dict` back or `None`.